File: Java/Language/Inheritance/AbstractClassTest/src/SavingsAccount.java
package edu.met.banking;

final class SavingsAccount extends Account {

	//a field declared with final modifier cannot be reassigned (constant)
	final static double MIN_BAL = 5000;

	SavingsAccount() {
		balance = MIN_BAL;
	}

	public void deposit(double amount) {
		balance += amount;
	}

	public void withdraw(double amount) throws InsufficientFundsException {
		if(balance - amount < MIN_BAL)
			throw new InsufficientFundsException();
		balance -= amount;
	}
}

File: dotnet/Language/Inheritance/InterfaceTest1/BankLib/SavingsAccount.cs
namespace Met.Banking;

sealed class SavingsAccount : Account, IProfitable
{
    const double MinBal = 5000;

    internal SavingsAccount()
    {
        Balance = MinBal;
    }

    public override void Deposit(double amount)
    {
        Balance += amount;
    }

    public override void Withdraw(double amount)
    {
        if(Balance - amount < MinBal)
            throw new InsufficientFundsException();
        Balance -= amount;
    }

    public double AddInterest(int months)
    {
        float rate = Balance < 20000 ? 4.0f : 4.5f;
        double interest = Balance * rate * months / 1200;
        Balance += interest;
        return interest;
    }
}
